### PR TITLE
Fixes #2926 - Notification bar (restart dialog) for crashes does not dismiss

### LINKED ISSUE #2926: Notification bar (restart dialog) for crashes does not dismiss
**Describe the issue you encountered:** Notification bar (restart confirmation) for crashes does not dismiss

**Expected behavior:**

- Platform (Win7, 8, 10? macOS? Linux distro?): Windows 10

- Brave Version: 0.11.2 RC1

- Steps to reproduce:
    1. Open about:preferences#advanced
    2. Change a pref that require restart to display a restart dialog
    3. process.crash()

- Screenshot if needed:
clipboard01

- Any related issues: #2907

### PR DESCRIPTION
**Pull request by willy-b**
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.

### CODE REVIEW

**Comment by luixxiul** [comment]
Wow, thanks for the quick patch!

**Comment by willy-b** [comment]
NP, thanks for running such an open project :-D

File: js/stores/appStore.js [hunk 1]
```diff
@@ -338,11 +338,16 @@ const handleAppAction = (action) => {
...
+
+        function notificationResponseCallback (e, message, buttonIndex, persist) {
           if (message === locale.translation('unexpectedErrorWindowReload')) {
             appActions.hideMessageBox(message)
+          } else {
+            ipcMain.once(messages.NOTIFICATION_RESPONSE, notificationResponseCallback)
           }
-        })
+        }
+        ipcMain.once(messages.NOTIFICATION_RESPONSE, notificationResponseCallback)
```
Comment (diracdeltas) [inline js/stores/appStore.js]: nice catch! may be simpler to change this to `ipcMain.on` and then call `ipcMain.removeListener`  when `message === locale.translation('unexpectedErrorWindowReload')`

**Comment by bbondy** [comment]
Thanks! I'll pull this in since I'm running RC builds now, but feel free to do a follow up for Yan's comment.

File: js/stores/appStore.js (hunk 1 above)
Comment (willy-b) [inline js/stores/appStore.js]: ditto! opened a PR to switch to ipcMain.removeListener and it is much cleaner now. thanks :-)

**Comment by willy-b** [comment]
Whoops, somehow missed notifications for this PR! (I always prefer web to apps, but I think I may need a GitHub app of some kind for notifications to keep up with Brave :-D)

The follow up PR for Yan's comment is open: #2995.

**Pull request merged**